server/routes/prediction.routes.js

```diff
@@ -366,131 +366,149 @@ function normalizeMatchFilter(filter, field){
+  Object.entries(filters).forEach(([field, data]) => {
+    if (data.matchMode === 'equals' && data.value) {
+      attributes.where[field] = data.value;
     }
-
-    // Handle date filtering
-    if (!filter.ignoreDateCutoff && !filter.filters?.solNum) {
-      if (configuration.getConfig('minPredictionCutoffDate')) {
-        attributes.where.date = { [Sequelize.Op.gt]: configuration.getConfig('minPredictionCutoffDate') };
-      } else if (configuration.getConfig('predictionCutoffDays')) {
-        const cutoff = new Date();
-        cutoff.setDate(cutoff.getDate() - configuration.getConfig('predictionCutoffDays'));
-        attributes.where.date = { [Sequelize.Op.gt]: cutoff };
-      }
-    }
-
-    if (filter.startDate) {
-      const start = Date.parse(filter.startDate);
-      const cutoff = Date.parse(configuration.getConfig('minPredictionCutoffDate', '1990-01-01'));
-      if (start > cutoff) {
-        attributes.where.date = {
-          ...attributes.where.date,
-          [Sequelize.Op.gt]: filter.startDate,
-        };
-      }
+  });
+  return attributes;
+};
+
+const applyDateFilters = (attributes, filter, config) => {
+  if (!filter.ignoreDateCutoff && !filter.filters?.solNum) {
+    if (config.getConfig('minPredictionCutoffDate')) {
+      attributes.where.date = { [Op.gt]: config.getConfig('minPredictionCutoffDate') };
+    } else if (config.getConfig('predictionCutoffDays')) {
+      const cutoff = new Date();
+      cutoff.setDate(cutoff.getDate() - config.getConfig('predictionCutoffDays'));
+      attributes.where.date = { [Op.gt]: cutoff };
     }
+  }
 
-    if (filter.endDate) {
+  if (filter.startDate) {
+    const start = Date.parse(filter.startDate);
+    const cutoff = Date.parse(config.getConfig('minPredictionCutoffDate', '1990-01-01'));
+    if (start > cutoff) {
       attributes.where.date = {
         ...attributes.where.date,
-        [Sequelize.Op.lt]: filter.endDate,
+        [Op.gt]: filter.startDate,
       };
     }
+  }
 
-    // Agency/Office filtering for non-admin users
-    if (!isGSAAdmin(user.agency, user.userRole)) {
-      // Apply agency and computed office filters directly
-      attributes.where.agency = user.agency;
+  if (filter.endDate) {
+    attributes.where.date = {
+      ...attributes.where.date,
+      [Op.lt]: filter.endDate,
+    };
+  }
+
+  return attributes;
+};
+
+const applyAgencyOfficeFilters = (attributes, user, filter) => {
+  if (!isGSAAdmin(user.agency, user.userRole)) {
+    const officeFilter = filter.filters?.office?.value;
+    
+    if (officeFilter) {
+      // Look for matches in both office and agency fields for the office value
+      attributes.where[Op.or] = [
+        { office: officeFilter },
+        { agency: officeFilter }  // This catches cases where EBUY data puts office in agency field
+      ];
+    } else {
+      // Default behavior - filter by user's office
       attributes.where.office = user.office;
```

Comment:
You are still filtering by user.office even though that will not exist

Reply:
Modified applyAgencyOfficeFilters and a few other places to remove any type of querying to a user's office(non-existant)